flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/ExecNodeContext.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.plan.nodes.exec;
+
+import org.apache.flink.table.planner.plan.nodes.exec.stream.StreamExecNode;
+
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonCreator;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonValue;
+
+/**
+ * Helper Pojo that holds the necessary identifier fields that are used for JSON plan serialisation
+ * and de-serialisation.
+ */
+public class ExecNodeContext {
+    /** The unique identifier for each ExecNode in the JSON plan. */
```

Comment:
Can you have these javadocs on the getters instead?

Reply:
I didn't do that, because the getters are not `@JsonProperty` annotated, instead the `toString()` is used for @JsonValue, the getters are only used in the IdResolver, and imho, makes more sense to keep the javadoc on the fields, but not insisting, happy to change if you prefer.

Reply:
I'm personally not a fan of the javadoc on the private fields, as when using the IDE, I won't see anything unless I navigate to the class source code. But as you prefer.

Reply:
moved to the getter, no problem!